src/Simulation/Simulators/QuantumSimulator/StateDumper.cs

```diff
@@ -60,29 +67,190 @@ public virtual bool Dump(IQArray<Qubit>? qubits = null)
+            /// </remarks>
+            [JsonProperty("amplitudes")]
+            public Complex[]? Amplitudes { get; set; }
+
+
+            // TODO(rokuzmin): Make this an extension method for `QuantumSimulator.DisplayableState` in IQSharp repo:
+            // /// <summary>
+            // ///     An enumerable source of the significant amplitudes of this state
+            // ///     vector and their labels, where significance and labels are
+            // ///     defined by the values loaded from <paramref name="configurationSource" />.
+            // /// </summary>
+            // public IEnumerable<(Complex, string)> SignificantAmplitudes(
+            //     IConfigurationSource configurationSource
+            // ) => SignificantAmplitudes(
+            //     configurationSource.BasisStateLabelingConvention,
+            //     configurationSource.TruncateSmallAmplitudes,
+            //     configurationSource.TruncationThreshold
+            // );
+
+            /// <summary>
+            ///     An enumerable source of the significant amplitudes of this state
+            ///     vector and their labels.
+            /// </summary>
+            /// <param name="convention">
+            ///     The convention to be used in labeling each computational basis state.
+            /// </param>
+            /// <param name="truncateSmallAmplitudes">
+            ///     Whether to truncate small amplitudes.
+            /// </param>
+            /// <param name="truncationThreshold">
+            ///     If <paramref name="truncateSmallAmplitudes" /> is <c>true</c>,
+            ///     then amplitudes whose absolute value squared are below this
+            ///     threshold are suppressed.
+            /// </param>
+            public IEnumerable<(Complex, string)> SignificantAmplitudes(
+                BasisStateLabelingConvention convention,
+                bool truncateSmallAmplitudes, double truncationThreshold
+            ) =>
+                (
+                    truncateSmallAmplitudes
+                    ? Amplitudes
+                        .Select((amplitude, idx) => (amplitude, idx))
+                        .Where(item =>
+                            System.Math.Pow(item.amplitude.Magnitude, 2.0) >= truncationThreshold
+                        )
+                    : Amplitudes.Select((amplitude, idx) => (amplitude, idx))
+                )
+                .Select(
+                    item => (item.amplitude, BasisStateLabel(convention, item.idx))
+                )
+                .OrderBy(
+                    item => item.Item2,
+                    // If a basis state label is numeric, we want to compare
+                    // numerically rather than lexographically.
+                    convention switch {
+                        BasisStateLabelingConvention.BigEndian => ToIntComparer,
+                        BasisStateLabelingConvention.LittleEndian => ToIntComparer,
```

Comment:
Would be nice to explain why both conventions result in the same comparer.

Reply:
This code has been moved from IQSharp repo too.
Based on the comments I assume that for numerical values (regardless of the endianness) the numerical comparer is used, but for the string values - the lexicographical comparer.
(@cgranade, please correct me if I'm wrong)

Reply:
Big-endian and little-endian are both used as different conventions for producing an int from a bitstring, but once you have an int, natural ordering on integers applies from there in both cases.

Reply:
And having this in a comment would be nice.

Reply:
I'm leaving this and a few more items as non-resolved. I will address those at stage 3 (see PR description).